pkg/k8s/identitybackend/identity.go

```diff
@@ -228,6 +230,20 @@ func (c *crdBackend) get(ctx context.Context, key allocator.AllocatorKey) *v2.Ci
 		return nil
 	}
 
+	sort.Slice(identities, func(i, j int) bool {
+		left, ok := identities[i].(*v2.CiliumIdentity)
+		if !ok {
+			return false
+		}
+
+		right, ok := identities[j].(*v2.CiliumIdentity)
+		if !ok {
+			return false
+		}
+
+		return left.CreationTimestamp.After(right.CreationTimestamp.Time)
```

Comment:
I think we should do `.Before()` so that we pick the first created identity, rather than the "duplicates" that came after, right?